flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/config/FlinkConnectorRateLimiter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.kafka.config;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.functions.RuntimeContext;
+
+/**
+ * An interface to create a ratelimiter
+ *
+ * <p>The ratelimiter is configured via {@link #setRate(long)} and
+ * created via {@link #create()}.
+ * An example implementation can be found {@link DefaultKafkaRateLimiter}.
+ * */
+
+@PublicEvolving
+public interface FlinkConnectorRateLimiter {
+
+	void open(RuntimeContext runtimeContext);
+
+	/** Creates a rate limiter.  */
+	<T> T create();
```

Comment:
This is confusing, the rate limiter was already created by the user.

Why is `acquire` not defined here?

Reply:
This does look confusing. Essentially, we need `runtimeContext` to be initialized to be able to determine the per-subtask rate. Perhaps, only the `open()` method should suffice (and all related logic can be implemented there)? 

As for `acquire()` not being here, that was my bad. I have fixed this to include `acquire()` and completely removed the Guava import in the `KafkaConsumerThread` class (which was not the case before).

Reply:
It looks like `create` is redundant. The work done in that method can be moved to `open`.

Reply:
Updated to only have `open()` and also moved the `create()` logic in the `GuavaFlinkConnectorRateLimiter` to reflect this.